command.go

```diff
@@ -5381,3 +5381,61 @@ func (cmd *InfoCmd) Item(section, key string) string {
 		return cmd.val[section][key]
 	}
 }
+
+type MonitorStatus int
+
+const (
+	MonitorStatusIdle MonitorStatus = iota
+	MonitorStatusStart
+	MonitorStatusStop
+)
+
+type MonitorCmd struct {
+	baseCmd
+	ch     chan string
+	status MonitorStatus
+}
+
+func NewMonitorCmd(ctx context.Context, ch chan string) *MonitorCmd {
+	return &MonitorCmd{
+		baseCmd: baseCmd{
+			ctx:  ctx,
+			args: []interface{}{"monitor"},
+		},
+		ch:     ch,
+		status: MonitorStatusIdle,
+	}
+}
+
+func (cmd *MonitorCmd) String() string {
+	return cmdString(cmd, nil)
+}
+
+func (cmd *MonitorCmd) readReply(rd *proto.Reader) error {
+	go cmd.readMonitor(rd)
```

Comment:
Can we do something with the following pattern maybe?

ctx, cancel := context.WithCancel(context.Background())
	go func(ctx context.Context) {
		for {
			select {
			case <-ctx.Done():
				fmt.Println("Goroutine has stopped!")
				return
			default:
				fmt.Println("This is the test goroutine")
				time.Sleep(time.Second * 3)
			}
		}
	}(ctx)

Then maybe we can have stop trigger a cancel() to properly shut down the goroutine? Alternatively - perhaps we can set a variable on the MonitorCmd struct and then check it. Stop would set the variable, the running routine would check and end accordingly -stop returning as usual